### DISCUSSION TITLE
Problems with structs in EEPROM on Pi PICO

### DISCUSSION THREAD

**Discussion by strud** [earlephilhower/arduino-pico#744] [Q&A] [answered]
Hello,

I have a project using a Pi PICO with a number of peripherals and libraries .

I am using the EEPROM emulator library to store configuration parameters.

The problem I'm having seems to be caused by the operation of my code (possibly the libraries) since when I use the same hardware with a minimalist EEPROM example withe structs, it works fine.

Is there a general approach I can take to prevent other background code (eg running in these various libraries) from interfering with the EEPROM emulator?  Does it already pause core execution and stop interrupts as required?

Are there any limits as to the size of the structure or the types within a structure that can be used?

example struct type I am using:

````
struct PadControllerConfigType {
   char name[10];
   unsigned int id;
   unsigned char loraAddrLO;
   unsigned char loraAddrHI;
   unsigned char lcLoraAddrLO;
   unsigned char loraChannel;
   float baseFreq;
   float channelFreq;
   unsigned char netID;
   unsigned int numberOfOutputs;
   char outputChannelNames[8];
   unsigned int numberOfInputs;
   char inputChannelNames[8];
   unsigned int numberOfAnalogInputs;
   AnalogChannelType analogInputConfigs[8];
   int txPower;
};
````

**Comment by strud** [comment, marked as answer]
Turns out to be a brain fart error on my part.

This does not work:

`void StoreParameters(void){
   int addr = EEPROM_BASE;
   EEPROM.put(addr, &padControllerConfig);
   EEPROM.commit();
}`

This does

`void StoreParameters(void){
   int addr = EEPROM_BASE;
   EEPROM.put(addr, padControllerConfig);
   EEPROM.commit();
}`

**Reply by earlephilhower** [reply]
Good debug.  These things happen to everyone!
